src/backend/executor/nodeShareInputScan.c

```diff
@@ -764,7 +857,8 @@ shareinput_reader_waitready(int share_id, PlanGenerator planGen)
 		tval.tv_usec = 0;
 
 		n = select(pctxt->readyfd+1, (fd_set *) &rset, NULL, NULL, &tval);
-
+	
+		timeout_count = share_input_scan_wait_tmpfile_timeout / tval.tv_sec;
```

Comment:
This is not with good design. If you want the timeout times, you could aggregate by checking n.